qemu/tests/hotplug_mem_share_discard_data.py

```diff
@@ -0,0 +1,93 @@
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.utils_numeric import normalize_data_size
+from virttest.utils_test.qemu import MemoryHotplugTest
+from virttest.staging import utils_memory
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory share and discard-data hotplug test
+    1) Setup hugepages
+    2) Mount /mnt/kvm_hugepage
+    3) Boot guest with one numa node
+    4) Hotplug 1G memory
+    5) Check memory
+    6) Unplug memory
+    7) Check memory
+    8) Check results for discard-data value
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = float(params.get("login_timeout", 240))
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    mem_name = params["target_mems"]
+    hp_size = utils_memory.read_from_meminfo("Hugepagesize")
+    hp_free = int(utils_memory.read_from_meminfo('HugePages_Free'))
+    size_target_mem = params["size_mem_%s" % mem_name]
+
+    hp_total = int(utils_memory.read_from_meminfo("HugePages_Total"))
+    hp_plugged = int(normalize_data_size(size_target_mem, "K"))
+    hp_target = int(hp_plugged / hp_size)\
+        + int(hp_total)
+    process.system("echo %s > /proc/sys/vm/nr_hugepages" % hp_target, shell=True)
```

Comment:
As we have discussed, it is better to leave hugepage setup to the framework instead of a simple echo.
I still suggest to use param `target_hugepages`,  however as it depends on the hugepage size,  define it in cfg is not a good choice (different env have different hugepage size), so i would like to suggest:
cfg:
```
not_preprocess = 'yes'
```

py:
```
    mem_dev = params.get('mem_devs') # only one mem_dev in this case, otherwise object_param
    size_mem =  int(normalize_data_size(params['size_mem_%s' % mem_dev], 'K'))
    total_hg_size = size_mem

    target_mem = params.get("target_mems") # only one target mem in this case
    if params.get('backend_mem_%s' % target_mem) == 'memory-backend-file':
        size_target_mem =  int(normalize_data_size(params['size_mem_%s' % target_mem], 'K'))
        total_hg_size += size_target_mem

    hp_size = utils_memory.read_from_meminfo("Hugepagesize")
    params['target_hugepages'] = int(total_hg_size // hp_size)
    params['setup_hugepages'] = "yes"
    params['not_preprocess'] = "no"

    env_process.preprocess(test, params, env)
```

Reply:
Then should I remove from cfg the param? :
```bash
setup_hugepages = yes
```

Reply:
Yes, it need be removed.